src/systemctl/systemctl-start-unit.c

```diff
@@ -399,3 +400,49 @@ int verb_start(int argc, char *argv[], void *userdata) {
 
         return ret;
 }
+
+int verb_toggle(int argc, char *argv[], void *userdata) {
+        _cleanup_strv_free_ char **names = NULL;
+        UnitActiveState active_state;
+        sd_bus *bus;
+        int r, i, j = 0;
+        bool is_active = false;
+        char *new_args[3];
+        static const UnitActiveState good_states[] = {
+                UNIT_ACTIVE,
+                UNIT_RELOADING,
+        };
+        int nb_states = ELEMENTSOF(good_states);
+
+        if (!argv[1])
+                return 0;
+
+        r = mangle_names("to toggle", strv_skip(argv, 1), &names);
+        if (r < 0)
+                return r;
+
+        r = acquire_bus(BUS_MANAGER, &bus);
+        if (r < 0)
+                return r;
+
+        STRV_FOREACH(name, names) {
+                is_active = false;
+
+                r = get_state_one_unit(bus, names[j], &active_state);
+                if (r < 0)
+                        return r;
+
+                for (i = 0; i < nb_states; ++i)
+                        if (good_states[i] == active_state)
+                                is_active = true;
```

Comment:
no need to loop, you can just use `is_active = IN_SET(active_state, UNIT_ACTIVE, UNIT_RELOADING)`

Reply:
thank you for pointing them out, the code is so much better! pushing new commit now.